polytope_server/common/datasource/fdb.py

```diff
@@ -51,7 +51,7 @@ def __init__(self, config):
         os.environ["FDB_HOME"] = self.config.get("fdb_home", "/opt/fdb")
         import pyfdb
```

Comment:
Why the import inside?

Reply:
pyfdb requires FDB to be installed in order to import but FDB is only one of the datasources. This is done to avoid making FDB a hard dependency

Reply:
what about having dynamic imports?

```
import importlib

class FDBDataSource(datasource.DataSource):
    def __init__(self, config):
        self.pyfdb = importlib.import_module("pyfdb")
        self.fdb = self.pyfdb.FDB()
```

Reply:
What is the advantage? Just to satisfy the linter?

Reply:
- Code readability and Intent: importlib explicitly communicate the intention to defer the dependency load until runtime. This makes it clear to maintainers and readers that this module isn't a core dependency

- Allow centralized handling for import failures

- Function-level imports are often seen as a "code smell" because: They make it harder to refactor or debug import errors. They aren't as idiomatic in Python, where module-level imports are preferred for clarity.

- Dynamic imports keep the module namespace cleaner. Unlike function-scoped imports, which can confuse static analysis tools, importlib imports are explicitly bound to a variable or attribute (self.pyfdb), reducing ambiguity in the module scope.

(opinion from the both :D) Should You Use It?
If your main concern is functionality and avoiding hard dependencies, function-level imports already satisfy that need. Switching to importlib offers mostly readability and maintainability improvements. It's not essential, but it aligns with best practices in dynamic module loading, especially in larger or more complex codebases.

If you don't encounter linter issues or readability concerns, sticking with the function-level import is perfectly fine.